src/Adapter/Product/SpecificPrice/Repository/SpecificPriceRepository.php

```diff
@@ -238,6 +248,53 @@ public function findExisting(
         return new SpecificPriceId($id);
     }
 
+    /**
+     * @param ProductId $productId
+     *
+     * @return PriorityList|null
+     */
+    public function findPrioritiesForProduct(ProductId $productId): ?PriorityList
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->select('spp.priority')
+            ->from($this->dbPrefix . 'specific_price_priority', 'spp')
+            ->where('spp.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+        ;
+
+        $result = $qb->execute()->fetchOne();
+
+        if (!$result) {
+            return null;
+        }
+
+        return new PriorityList(explode(';', $result));
+    }
+
+    /**
+     * @return PriorityList
+     *
+     * @throws CoreException
+     */
+    public function getDefaultPriorities(): PriorityList
+    {
+        try {
+            $priorities = SpecificPrice::getPriority(0);
```

Comment:
You should use the configuration value directly here The `SpecificPrice::getPriority` performs an SQL query first but you don't need this here so you can get the `PS_SPECIFIC_PRICE_PRIORITIES` via the configuration service and explode it